### PR TITLE
[GLUTEN-2854] [VL] Refactor native defaultLeafVeloxMemoryPool usage for test and validation only

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

This patch limit defaultLeafPool method's usage, it only used in test and validation now.

After this patch, all memory pools used in real computation were tracked by Spark.

Fixes: #2854

### CODE REVIEW

**Comment by Yohahaha** [comment]
@zhztheplayer could you help take a look? thanks!

File: cpp/velox/memory/VeloxMemoryManager.h [hunk 1]
```diff
@@ -54,7 +54,16 @@ class VeloxMemoryManager final : public MemoryManager {
...
+  static auto memoryManager = std::make_shared<gluten::VeloxMemoryManager>(
+      "test",
+      gluten::defaultMemoryAllocator(),
+      std::shared_ptr<gluten::AllocationListener>(gluten::AllocationListener::noop()));
+  return memoryManager;
+}
+
+inline std::shared_ptr<facebook::velox::memory::MemoryPool> defaultLeafVeloxMemoryPool() {
+  return getDefaultMemoryManager()->getLeafMemoryPool();
+}
```
Comment (zhztheplayer) [inline cpp/velox/memory/VeloxMemoryManager.h:68]: Why we inline the functions? They should not be called very frequently.

**Review by zhztheplayer — COMMENTED** [review]
Only a minor comment. Thanks!

File: cpp/velox/memory/VeloxMemoryManager.h (hunk 1 above)
Comment (Yohahaha) [reply cpp/velox/memory/VeloxMemoryManager.h:68]: These codes were moved from BenchmarkUtils.h.

> Why we inline the functions?

Good question, and I can not give a reasonable answer, just follow other codes.

Comment (Yohahaha) [reply cpp/velox/memory/VeloxMemoryManager.h:68]: Oh I see the origin codes was added by me... 

Inline or not are seems good to me.

**Review by zhztheplayer — APPROVED** [review]

**Pull request merged**